cashctrl_ledger/tests/test_ledger.py

```diff
@@ -1,124 +1,171 @@
 
         # Updating a ledger with non existent tax code should raise an error
         target = ledger_entry.copy()
         target["id"] = id
         target["tax_code"] = "Test_Non_Existent_TAX_code"
         with pytest.raises(ValueError, match="No id found for tax code"):
-            ledger.modify_ledger_entry(target)
+            restored_engine.ledger.modify(target)
 
         # Updating a ledger with non existent account code should raise an error
         target = ledger_entry.copy()
         target["id"] = id
-        target.loc[0, "account"] = 333333
+        target["account"] = 333333
         with pytest.raises(ValueError, match="No id found for account"):
-            ledger.modify_ledger_entry(target)
+            restored_engine.ledger.modify(target)
 
         # Updating a ledger with non existent currency code should raise an error
         target = ledger_entry.copy()
         target["id"] = id
-        target.loc[0, "currency"] = "Non_Existent_Currency"
+        target["currency"] = "Non_Existent_Currency"
         with pytest.raises(ValueError, match="No id found for currency"):
-            ledger.modify_ledger_entry(target)
+            restored_engine.ledger.modify(target)
 
         # Delete the ledger entry created above
-        ledger.delete_ledger_entries([id])
+        restored_engine.ledger.delete([{"id": id}])
 
-    def test_update_non_existent_ledger(self, ledger):
-        target = self.LEDGER_ENTRIES.query("id == '1'").copy()
-        target["id"] = 999999
-        with pytest.raises(RequestException):
-            ledger.modify_ledger_entry(target)
+    @pytest.mark.skip(reason="We don't have mechanism to allow missing ids")
+    def test_delete_entry_allow_missing(self):
+        pass
 
-    def test_delete_non_existent_ledger(self, ledger):
-        with pytest.raises(RequestException):
-            ledger.delete_ledger_entries(ids=["non-existent"])
+    def test_delete_nonexistent_entry_raise_error(self, restored_engine):
+        with pytest.raises(RequestException, match="API call failed. ID missing."):
+            restored_engine.ledger.delete({"id": ["FAKE_ID"]})
 
-    def test_split_multi_currency_transactions(self, ledger):
+    def test_adding_transaction_with_two_non_reporting_currencies_fails(self, restored_engine):
+        LEDGER_CSV = """
+            id,   date, account, contra, currency,    amount, reporting_amount, text, description
+            0, 2024-06-26,     ,   1000,      CHF, 100000.00,         111476.00, Convert CHF to EUR,
+            0, 2024-06-26, 1005,       ,      EUR,  93750.00,         90000.00, Convert CHF to EUR,
+            1, 2024-06-26,     ,   1000,      USD, 200000.00,        180000.00, Convert USD to EUR+CHF,
+            1, 2024-06-26, 1005,       ,      EUR,  93750.00,         90000.00, Convert USD to EUR+CHF,
+            1, 2024-06-26, 1010,       ,      CHF,  90000.00,         90000.00, Convert USD to EUR+CHF,
+        """
```

Comment:
We have a ledger entry with multiple currencies in our test data. 
Why not use that?

Reply:
That transactions are skipped now because of an error, I'll add a TODO to replace them when issues are resolved.